File: sorting-algorithms/Javascript/selectionsort.js
function selectionSort(arr){
    let n = arr.length; //The number of elements in array
    for(let i = 0; i < n-1; i++){
        //i is the number of passes
        console.log(`Pass ${i}`)
        console.log(`Index of unsorted array ${i}`)
        let minIndex = i; //Assume the index of the minimum element be the 0 (The first element is the minimum)
        for (let j = i+1; j < n; j++){
            console.log(`Next index to check: ${j}`)
            console.log(`Minimum index ${minIndex}`)
            if (arr[j] < arr[minIndex]){
                console.log(`New minimum element: ${arr[j]} at index ${j}`)
                //Let the minIndex be j since the element at j is less than minIndex
                minIndex = j;
            }

            console.log(`Before swap ${arr}`)

            //Swap the items to their appropriate position
            let temp = arr[minIndex];
            arr[minIndex] = arr[i];
            arr[i] = temp

            console.log(`After swap ${arr} \n`)
        }
    }

    return arr
}

let arr = [6,5,4,3,2,1]
console.log(`Before sorting: ${arr}`)
// console.time("start")
let sortedArr = selectionSort(arr);
// console.timeEnd("start")
console.log(`After sorting: ${sortedArr}`)

File: sorting-algorithms/Python/selectionsort.py
def find_smallest(arr):
    smallest_index = 0
    smallest_item = arr[smallest_index]

    for i in range(1, len(arr)):
        if (arr[i] < smallest_item):
            smallest_index = i
            smallest_item = arr[smallest_index]
    return smallest_index

def selection_sort(arr):
    sorted_arr = []
    for i in range(len(arr)):
        smallest = find_smallest(arr)
        sorted_arr.append(arr.pop(smallest))
    return sorted_arr


my_arr = [10,3,2,5,6,1]
sorted_arr = selection_sort(my_arr)

print(sorted_arr)
